types/video.js/index.d.ts

```diff
@@ -15,285 +15,7069 @@
 // Javascript, whether the browser is playing the video through HTML5
 // video, Flash, or any other supported playback technologies.
 
-declare function videojs(
-    id: any,
-    options?: videojs.PlayerOptions,
-    ready?: () => void
-): videojs.Player;
-export default videojs;
-export as namespace videojs;
+declare module 'video.js' {
```

Comment:
Since this file is `video.js/index.d.ts`, it should be unnecessary to use `declare module`, and you should be able to just `export default vjsObj` at the top-level.

Reply:
Will fix that soon. Thanks

Reply:
[module augmentation](https://www.typescriptlang.org/docs/handbook/declaration-merging.html) is indeed confusing. I think the problem you are probably running into is that the plugin file contains `declare module 'video.js' { ... }` and nothing else. If that happens, the file is in *global* scope and declares a new module. To augment a module you need to already be in the scope of another external module -- this can be done by writing `import {} from "video.js";` at the top.

Reply:
Yep, that was the missing piece of the puzzle. Already fixed! thanks again ;)